lib/THNN/generic/SpatialMaxPooling.c

```diff
@@ -61,18 +61,8 @@ static void nn_(SpatialMaxPooling_updateOutput_frame)(real *input_p, real *outpu
   }
 }
 
-static int nn_(SpatialMaxPooling_updateOutput)(lua_State *L)
+void THNN_(SpatialMaxPooling_updateOutput)(THNNState *state, THTensor *input, THTensor *output, THTensor *indices, int kW, int kH, int dW, int dH, int padW, int padH, int ceil_mode)
```

Comment:
Historically, [`nn` was C89](https://github.com/torch/nn/pull/2#issuecomment-31919246) and there is no `bool` in C89, but that has changed and [since this commit](https://github.com/torch/nn/pull/550) `nn` is officially C99.
Plus, `THNN` already support the [`bool` macro](https://github.com/torch/nn/blob/master/lib/THNN/THNN.h#L4), so I should probably update it.

Regarding the indices being float, long ago [they were actually `long` but were moved to `float` to work on the gpu](https://github.com/torch/nn/pull/46#issuecomment-49786258), but with the addition of other tensor types to cutorch, we can soon start thinking about updating it.